### PR TITLE
Update code, tutorials, and code snippets to be compatible with async circuits

### LINKED ISSUE #883: Update tutorials ,example code, and code snippets to be compatible with async circuits
Async circuits. Require all smart contract and zkprogram methods to be async https://github.com/o1-labs/o1js/pull/1477

Update all tutorial code,  example code, and code snippets.

- [ ]  update https://github.com/methods to be async
- [ ]  await to all contract calls await MyContract.myMethod();
- [ ]  Use the new @method.returns() decorator to declare a return value from a method

### PR DESCRIPTION
**Pull request by ymekuria**
# Description 

Closes #883

This PR updates all example code and example snippets with async circuits  following  the latest release of version `0.18.0` of `o1js`. https://github.com/o1-labs/o1js/pull/1543

### CODE REVIEW

**Review by MartinMinkov — APPROVED** [review]

File: examples/zkapps/02-private-inputs-and-hash-functions/package.json [hunk 1]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "0.15.*"
+      "o1js": "0.18.*"
```
Comment (MartinMinkov) [inline examples/zkapps/02-private-inputs-and-hash-functions/package.json:46]: Unnecessary indent?

File: examples/zkapps/05-common-types-and-functions/package.json [hunk 2]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "0.15.*"
+  "o1js": "0.18.*"
```
Comment (MartinMinkov) [inline examples/zkapps/05-common-types-and-functions/package.json:46]: indent changed here?

File: examples/zkapps/08-custom-tokens/package.json [hunk 3]
```diff
@@ -43,6 +43,6 @@
     "typescript": "^4.7.2"
   },
   "peerDependencies": {
-    "o1js": "0.15.*"
+    "o1js": "1.0.*"
```
Comment (MartinMinkov) [inline examples/zkapps/08-custom-tokens/package.json:46]: this is `1.0.*` but others are `0.18.*`?

File: examples/zkapps/02-private-inputs-and-hash-functions/package.json (hunk 1 above)
Comment (ymekuria) [reply examples/zkapps/02-private-inputs-and-hash-functions/package.json:46]: I'll update.

File: examples/zkapps/08-custom-tokens/package.json (hunk 3 above)
Comment (ymekuria) [reply examples/zkapps/08-custom-tokens/package.json:46]: This example has actually been removed from the current tutorials and Mina Foundation is working on a new one. Either way I will update this. Thanks!

**Pull request merged**